wireapp/wire-ios-mono
# fix: add recovery step for database migration failure WPB-3313

<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-3313" title="WPB-3313" target="_blank"><img alt="Bug" src="https://wearezeta.atlassian.net/rest/api/2/universal_avatar/view/type/issuetype/avatar/10803?size=medium" />WPB-3313</a>  Missing recovery step when database migration error occurs
  </td></table>
  <br />
 






















----
#### PR Submission Checklist for internal contributors

- The **PR Title**

- The **PR Description**
----

# What's new in this PR?

### Issues

When database migration fails the user is presented with alert to open settings or delete database files. When user selects option of "open settings" and get back to the app, he is stuck on the shield page.

### Causes (Optional)

The alert is not presented again when getting to the app from background.

### Solutions

1. Show migration failure alert again when app returns from background
2. Add "Retry" action to try migrating database once more

### Testing

#### How to test

1. Log in into app
2. Use Shake Gesture
3. Open `Configure flags` menu
4. Turn `forceDatabaseLoadingFailure` switch on
5. Restart the app
6. The database migration failure alert should be presented

#### Test Coverage (Optional)


----
#### PR Post Submission Checklist for internal contributors (Optional)

----
#### PR Post Merge Checklist for internal contributors

----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.
